onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -2494,9 +2494,12 @@
 };
 
 void GatherOpBuilder::AddInitializersToSkip(ModelBuilder& model_builder, const NodeUnit& node_unit) const {
-  // Skip the second input `indices` for Gather
   const auto& inputs = node_unit.Inputs();
-  model_builder.AddInitializerToSkip(inputs[1].node_arg.Name());  // indices
+  const auto& indices_name = inputs[1].node_arg.Name();
+  if (Contains(model_builder.GetInitializerTensors(), indices_name)) {
+    // Skip the second input `indices` for Gather if it is an initializer
+    model_builder.AddInitializerToSkip(indices_name);
+  }
 }
```

Comment:
Should we be skipping it if it's int32? In that case can we use the initializer as is? 

If so here and line 2522 should check the type.

Reply:
yeah, thanks for the catch up. it should skip for all non-int32 initializers here.

updated it. also reverted some changes in gather UT.